client-data/tools/ellipse/ellipse.js

```diff
@@ -142,6 +142,11 @@
         shape.cy.baseVal.value = Math.round((data['y2'] + data['y']) / 2);
         shape.rx.baseVal.value = Math.abs(data['x2'] - data['x']) / 2;
         shape.ry.baseVal.value = Math.abs(data['y2'] - data['y']) / 2;
+        if (data.deltax != undefined  ||  data.deltay != undefined) {
+            var tmatrix = Tools.getTranslateMatrix(shape);
+            tmatrix.e = data.deltax ||0;
+            tmatrix.f = data.deltay ||0;
+        }
```

Comment:
The goal of moving to a transform was to avoid having the tools being aware one of another. The transformation  code should probably not be duplicated in all tools, but should remain in the transformation tool's code.

Reply:
I am not sure how you'd like it, let's focus in one shape: the rectangle.

After implementing the move using translations rectangles, in addition of `x`, `y`, `x2`, `y2`, `stroke` ..., also have a an extra coordinate containing how much they have been moved.

To properly build a rectangle I need to set up this extra coordinate, this is what the piece of code underlined does.
That piece of code is similar in all shapes, because all shapes need to be constructed using those extra coordinates, just like to set up the `x` and `y` is similar in all shapes.

However, the moving code (how the extra coordinate is computed) is inside the `moving.js` file, the new function `Tools.getTranslateMatrix` gets (or make if not there) the transformation matrix of an svg object.

Reply:
Yes, I understand that the shape has to be created with the right translation. I am just saying that the code doing that should not be duplicated in every tool, but should instead invoked after the element has been created, without having to change the tool's code.

Reply:
That would be possible if the tools store the new `deltax` and `deltay` values somewhere. On loading the move tool can reposition everything in the right place.

Is the raw json data, and in particular the `deltax` and `deltay`, stored somewhere?

Reply:
No, unfortunately it's not. There is not real data model, and on the client side, the data is stored only in the DOM.

Reply:
I had an idea I am working on now. Tell me if it is acceptable.

Basically, in the function `messageForTool` in `js/board.js` I detect if the message has an `id` and a `deltax` or a `deltay`. In the case I send a message to the mover to reposition that shape.

It pushes a bit of the mover logic in the `board.js`, but at least it will be limited in one function that would continue working even if the mover tool is disabled.

Reply:
Yes, this sounds good to me 👍

Reply:
Please have a look